src/perl.c

```diff
@@ -33,6 +33,10 @@
 /* do not automatically get the thread specific Perl interpreter */
 #define PERL_NO_GET_CONTEXT
 
+#if defined(__APPLE__)
+#pragma clang diagnostic ignored "-Wcompound-token-split-by-macro"
+#endif
```

Comment:
According to:
* https://reviews.llvm.org/D86751
* https://github.com/Perl/perl5/issues/18780

It's a **new** option that came with Clang-12 around 2021.

Also, Clang is not Apple specific, and one could be using GCC on Apple.

=> I think this should be changed to ignore the diagnostic for Clang >= 12, not for Apple.